packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -152,116 +150,58 @@ function tryInferInputContext(change: OptionalChangeset): "empty" | "filled" | u
-		};
-
-		let earliestReservedDetachId: RegisterId | undefined;
-		let inputContext: "empty" | "filled" | undefined;
-
-		const childChangesByOriginalId = new RegisterMap<TaggedChange<NodeChangeset>[]>();
-		// TODO: It might be possible to compose moves in place rather than repeatedly copy.
-		// Additionally, working out a 'register allocation' strategy which enables frequent cancellation of noop moves
-		// for sandwich rebases would help with cloning if in-place proves too difficult
-		const current = getBidirectionalMaps([]);
-		for (const { change, revision } of changes) {
-			inputContext ??= tryInferInputContext(change);
-			const withIntention = (id: RegisterId): RegisterId => {
-				if (id === "self") {
-					return id;
-				}
-				const intention = id.revision ?? revision;
-				return { revision: intention, localId: id.localId };
-			};
-
-			const nextSrcToDst = new RegisterMap<
-				[dst: RegisterId, target: "nodeTargeting" | "cellTargeting"]
-			>();
-			const nextDstToSrc = new RegisterMap<RegisterId>();
-
-			if (change.reservedDetachId !== undefined && earliestReservedDetachId === undefined) {
-				earliestReservedDetachId = withIntention(change.reservedDetachId);
-			}
-
-			// Compose all the things that `change` moved.
-			for (const [unintentionedSrc, unintentionedDst, target] of change.moves) {
-				const src = withIntention(unintentionedSrc);
-				const dst = withIntention(unintentionedDst);
-				let originalSrc = current.dstToSrc.get(src);
-				let currentTarget: "cellTargeting" | "nodeTargeting" | undefined;
-				if (originalSrc !== undefined) {
-					const [dst2, existingTarget] =
-						current.srcToDst.get(originalSrc) ?? fail("expected backward mapping");
-					assert(
-						areEqualRegisterIds(dst2, src),
-						0x855 /* expected consistent backward mapping */,
-					);
-					currentTarget = existingTarget;
-				} else {
-					originalSrc = src;
-				}
-				nextSrcToDst.set(originalSrc, [
-					dst,
-					target === "cellTargeting" &&
-					(currentTarget === undefined || currentTarget === "cellTargeting")
-						? "cellTargeting"
-						: "nodeTargeting",
-				]);
-				nextDstToSrc.set(dst, originalSrc);
+		const inputContext = tryInferInputContext(change1) ?? tryInferInputContext(change2);
+
+		const earliestReservedDetachId: RegisterId | undefined =
+			withRevisionOrUndefined(change1.reservedDetachId, revision1) ??
+			withRevisionOrUndefined(change2.reservedDetachId, revision2);
+
+		const { srcToDst, dstToSrc } = getBidirectionalMaps(change1.moves, revision1);
+		for (const [src, dst, target] of change2.moves) {
+			const srcWithRevision = withRevision(src, revision2);
+			const dstWithRevision = withRevision(dst, revision2);
+			const originalSrc = dstToSrc.get(srcWithRevision);
+			if (originalSrc !== undefined) {
+				const entry = srcToDst.get(originalSrc);
+				assert(entry !== undefined, "There should be a corresponding entry");
+				entry[0] = dstWithRevision;
+			} else {
```

Comment:
This changes the original behavior in the following way:
If `entry[1]` is `"cellTargetting"` and `target` is `"nodeTargeting"`, then the old code produced a `"nodeTargeting"` move while the new code produces a `"cellTargetting"` move.
This was clearly not test-enforced, but I'm guessing you din't intend to change this.

Reply:
I don't really understand the logic of this, but I've changed it to try to match the original behavior.